src/pgduckdb_types.cpp

```diff
@@ -1505,11 +1507,16 @@ AppendString(duckdb::Vector &result, Datum value, idx_t offset, bool is_bpchar)
 
 static void
 AppendJsonb(duckdb::Vector &result, Datum value, idx_t offset) {
+	D_ASSERT(duckdb_pg_scan_memory_ctx != nullptr);
+	MemoryContext old_context = MemoryContextSwitchTo(duckdb_pg_scan_memory_ctx);
 	auto jsonb = DatumGetJsonbP(value);
-	auto jsonb_str = JsonbToCString(NULL, &jsonb->root, VARSIZE(jsonb));
-	duckdb::string_t str(jsonb_str);
+	StringInfo str = makeStringInfo();
+	auto json_str = JsonbToCString(str, &jsonb->root, VARSIZE(jsonb));
 	auto data = duckdb::FlatVector::GetData<duckdb::string_t>(result);
-	data[offset] = duckdb::StringVector::AddString(result, str);
+	data[offset] = duckdb::StringVector::AddString(result, json_str, str->len);
+	pfree(str->data);
+	pfree(str);
```

Comment:
Is this still needed? If not, let's keep this simple and just rely on the memory context resets.